### DISCUSSION TITLE
weight parameter of DiceCELoss

### DISCUSSION THREAD

**Discussion by NastaranVB** [Project-MONAI/MONAI#7522] [Q&A]
Hello,
I am doing multi class segmentation which contains 3 classes including background. I want to consider weight parameter of "DiceCEloss", so have defined loss function as below:
loss_function=DiceCELoss(include_background=False, to_onehot_y=True, softmax=True, weight=torch.tensor([1, 10])).
I get the below error:
ValueError: the length of the `weight` sequence should be the same as the number of classes.
                        If `include_background=False`, the weight should not include
                        the background category class 0.

Thank you in advance.

**Comment by KumoLiu** [comment]
Hi @NastaranVB, please set the weight contain background; this will remove the background class internal. Because weight must also be used in 'CrossEntropyLoss'.
https://github.com/Project-MONAI/MONAI/blob/e9e273858b6904659b0cd581d357662dc5c782f6/monai/losses/dice.py#L714

Thanks.

**Reply by NastaranVB** [reply]
@KumoLiu  thanks for coming back to me so quickly. 
So, it should be as below, am I right?
loss_function=DiceCELoss(include_background=False, to_onehot_y=True, softmax=True, weight=torch.tensor([1., 2., 10.])).
And now the weight is applied for both 'DiceLoss' and 'CrossEntropyLoss'?

As an another question, let me ask what is the parameter "ce_weight=None" in definition of class "DiceCELoss"? Is it weight related to 'CrossEntropyLoss'? How should it be set? 
I just got a bit confused, I would be thankful if you could please explain a bit how I should set "ce_weight' and "weight".

Thank you in advance.

**Reply by KumoLiu** [reply]
Hi @NastaranVB, it's a deprecated argument which will be removed in 1.4.
https://github.com/Project-MONAI/MONAI/blob/e9e273858b6904659b0cd581d357662dc5c782f6/monai/losses/dice.py#L650

Thanks.
